src/interpreter/mod.rs

```diff
@@ -244,6 +246,8 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
     }
 
     fn type_is_sized(&self, ty: Ty<'tcx>) -> bool {
+        // generics are weird, don't run this function on a generic
+        debug_assert_eq!(self.monomorphize(ty, self.substs()), ty);
```

Comment:
You can check this with `ty.needs_substs()` which is basically free.
IMO `debug_assert` is bad in the compiler (compared to always-on cheap checks) because users won't trigger it.